rcl/src/rcl/expand_topic_name.c

```diff
@@ -167,20 +167,16 @@ rcl_expand_topic_name(
           *output_topic_name = NULL;
           char * unmatched_substitution =
             rcutils_strndup(next_opening_brace, substitution_substr_len, allocator);
-          char * allocated_msg = NULL;
-          char * msg = NULL;
           if (unmatched_substitution) {
-            allocated_msg = rcutils_format_string(
+            RCL_SET_ERROR_MSG_WITH_FORMAT_STRING(
               allocator,
               "unknown substitution: %s", unmatched_substitution);
-            msg = allocated_msg;
           } else {
-            SAFE_FWRITE_TO_STDERR("failed to allocate memory for error message\n");
-            msg = "unknown substitution: allocation failed when reporting error";
+            SAFE_FWRITE_TO_STDERR("failed to allocate memory for unmatched substitution\n");
+            RCL_SET_ERROR_MSG("unknown substitution: allocation failed when reporting error",
+              allocator);
```

Comment:
These two lines seem redundant. What are the chances for the second statement to work if memory allocation is failing?

Reply:
we can simplify it to only indicate the allocation 
failure from `rcutils_strndup()`.